apps/nextjs-website/src/components/organisms/GitBookContent/components/Expandable.tsx

```diff
@@ -56,12 +56,32 @@ const AccordionDetails = styled(MuiAccordionDetails)(({ theme }) => ({
   paddingRight: theme.spacing(2),
 }));
 
-const Expandable = ({ children }: ExpandableProps<ReactNode>) =>
-  children && (
-    <Accordion disableGutters variant='outlined' sx={{ my: 2 }}>
-      {children}
-    </Accordion>
+const Expandable = ({ children }: ExpandableProps<ReactNode>) => {
+  // if children is array and an item is string wrap it with paragraph
+  if (Array.isArray(children)) {
+    // eslint-disable-next-line no-param-reassign
+    children = children.map((child, index) =>
+      typeof child === 'string' ? (
+        <Paragraph key={index}>{child}</Paragraph>
+      ) : (
+        child
+      )
+    );
+  }
```

Comment:
What exactly does this 'if' statement resolve? Could we potentially address this at the parsing level instead?

Reply:
Expandable supports both a react component or a single string. This wraps the single string with a paragraph so that it is rendered correctly because it would not be correct to wrap a react node with a paragraph (mui also does not allow this behaviour)

Reply:
Digging deeper into what led to adding this `if` statement, I concluded that the root cause is the [workaround](https://github.com/pagopa/developer-portal/blob/b2dedb0/packages/gitbook-docs/src/parseContent.ts#L96-L109) used to parse HTML tags in markdown files. The [xml details](https://github.com/pagopa/devportal-docs/blob/63a1d8c/docs/PiBduZS70S7Ae4WpWywl/appendici/primitive.md?plain=1#L1997) tag is transformed into a Markdoc tag, parsed as a [details](https://github.com/pagopa/developer-portal/blob/b2dedb0/packages/gitbook-docs/src/parseContent.ts#L67) and rendered as Expandable element, which is producing unexpected results:

![Screenshot 2024-03-14 at 10 15 27](https://github.com/pagopa/developer-portal/assets/4775679/c1d41f0c-b839-46cf-a953-1e99885ed624)
> Check out it visiting the page http://localhost:3000/pago-pa/guides/sanp/appendici/primitive expanding the `sendPaymentOutcome`

To fix the problem, I suggest the following steps:
1. Eliminate the workaround by removing [this line](https://github.com/pagopa/developer-portal/blob/b2dedb0/packages/gitbook-docs/src/parseContent.ts#L106) and [this line](https://github.com/pagopa/developer-portal/blob/b2dedb0/packages/gitbook-docs/src/parseContent.ts#L49)
2. Add the HTML tag parsing by replacing [this line](https://github.com/pagopa/developer-portal/blob/b2dedb0/packages/gitbook-docs/src/parseContent.ts#L67) with `htmldetails: details`
3. Restore the original Expandable react component

After that the `xml` response is rendered correctly:
![Screenshot 2024-03-14 at 10 27 53](https://github.com/pagopa/developer-portal/assets/4775679/ac42ea7b-95e5-4dcf-bf3f-563a8dccc521)